### ISSUE TITLE
Improving `units::Duration`

### ISSUE THREAD

**Issue by elBoberido** [eclipse-iceoryx/iceoryx#536]
## Brief feature description

The `units::Duration` class could have same additional features and improvements.

## Detailed information

- [x] rename the creation methods like `milliseconds` to `fromMilliseconds`
    - some getter methods should then be renamed, e.g. `milliSeconds` -> `milliseconds` or `toMilliseconds`
    - no final decision, please ask before starting with this
- [ ] conversion to/from all `chrono` types
- [x] `operator+=`, `operator-=` and `operator*=` https://github.com/eclipse-iceoryx/iceoryx/pull/501#discussion_r564463012
- [ ] abstractions for `TimePoint` for monotonic and system clock https://github.com/eclipse-iceoryx/iceoryx/pull/501#discussion_r560061177
    - needs to be discussed and maybe another issue should be used for this

**Comment by enkeyz** [comment]
Could work on the renaming(first one), if it's still wanted.

**Comment by elfenpiff** [comment]
@enkeyz It is still wanted and we would very much appreciate it!

I would assign the issue to you and if you run into troubles or other issues just contact us via https://gitter.im/eclipse/iceoryx

**Comment by elBoberido** [comment]
@enkeyz sure, go ahead. If it's not too late, please rename the getter methods with a `to` prefix, so `toDays`, `toNanoseconds`. This makes it's less ambiguous than just `nonoseconds` which could also mean just to nanoseconds part of the duration.

**Comment by elfenpiff** [comment]
> Already finished with it, just one question: how to correctly run the "run_all_tests.sh" from tools folder? 

@enkeyz nice!

> Getting error message: "make: *** No rule to make target 'all_tests'. Stop."

I would recommend to use the iceoryx_meta approach for developing. Here you can compile and run the tests with the following commands:

```sh
cd iceoryx
cmake -Bbuild -Hiceoryx_meta -DBUILD_TEST=ON
cd build
make
make all_tests # executes all unit- and integrationtests
```
